suite-common/wallet-core/src/discovery/discoveryThunks.ts

```diff
@@ -702,8 +702,34 @@ export const runAdditionalDiscoveryThunk = createThunk(
 
         TrezorConnect.on<DiscoverAccountsProgress>(UI.BUNDLE_PROGRESS, onBundleProgress);
 
+        // NOTE: prepare the device to the corresponding state eg. insert the passphrase
+        const stateResult = await TrezorConnect.getDeviceState({
+            device: {
+                path: device.path,
+                instance: device.instance,
+                state: device.state.staticSessionId,
+            },
+            useEmptyPassphrase: device.useEmptyPassphrase,
+        });
+
+        if (stateResult.success) {
+            dispatch(
+                applyDeviceStatesThunk({
+                    newDeviceState: stateResult.payload._state,
+                    isAddingHiddenWallet: !device.useEmptyPassphrase,
+                    devicePath: device.path,
+                }),
+            );
+        }
+
         const result = await TrezorConnect.discoverAccounts({
-            device,
+            device: {
+                ...device,
```

Comment:
I think you would need to call a selector again to get the freshly updated device.